tests/tests/test_guest_cpu_off.py

```diff
@@ -67,39 +55,16 @@ def test_guest_cpu_pinned_off():
         with Qemu.QemuMachine() as qm:
             qm.run_and_wait()
 
-            cpu = pin_process_on_random_cpu(qm.pid)
+            cpu = util.cpu_select()
 
-            cpu_on_off(f'/sys/devices/system/cpu/cpu{cpu}/online', 0)
+            with util.CpuOnOff(cpu) as cpu_manager:
+                # bring the cpu online if necessary
+                cpu_manager.set_state(1)
+                util.pin_process_on_cpu(qm.pid, cpu)
 
-            m = Qemu.QemuSSH(qm)
-            m.check_exec('sudo init 0 &')
+                # bring the cpu core offline
+                cpu_manager.set_state(0)
+                m = Qemu.QemuSSH(qm)
+                m.check_exec('sudo init 0 &')
```

Comment:
Small nit, I think its worth explicitly setting state back to 1 even if the context manager sets it .  When someone looks at this code they may not understand what is happening when the object gets destroyed at first and personally, I prefer explicit and then using implicit (context manager) as a backup.

Reply:
@spmcmillan  I see, however, we do not want to set the core state to 1, but to the previous state before the test (for example, the core can be offline just before the test), the context manager does it for us right now.

Reply:
I think you are saying its possible the cpu came in turned off already.  Which I suppose is true